src/Sarif/Baseline/ResultMatching/DataStructures/ExtractedResult.cs

```diff
@@ -15,7 +12,68 @@ namespace Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching
     public class ExtractedResult
     {
         public Result Result { get; set; }
-        
         public Run OriginalRun { get; set; }
+        public string RuleId { get; set; }
+
+        public ExtractedResult(Result result, Run run)
+        {
+            Result = result;
+            OriginalRun = run;
+
+            // Look up and cache the RuleId
+            RuleId = result.ResolvedRuleId(run);
+        }
+
+        /// <summary>
+        ///  Match the 'Category' of two ExtractedResults (Tool and RuleId)
+        /// </summary>
+        /// <param name="other">ExtractedResult to match</param>
+        /// <returns>True if Category is identical, False otherwise</returns>
+        public bool MatchesCategory(ExtractedResult other)
+        {
+            return this.RuleId == other.RuleId;
+
+            // Tool contributes to category, but SarifLogMatcher ensures only Runs with matching Tools are compared,
+            // so we don't check here.
+        }
+
+        /// <summary>
+        ///  Match enough of the 'What' properties of two ExtractedResults (Guid, Fingerprints, PartialFingerprints, Snippets, Message, Properties).
+        ///  A match in high-confidence identity properties is a match (Guid, Fingerprint, >= 50% of PartialFingerprint).
+        ///  A non-match in high-confidence identity properties is a non-match (Fingerprint, 0% of PartialFingerprints, Properties).
+        ///  Otherwise, Results match if Message and first Snippet match.
+        /// </summary>
+        /// <param name="other">ExtractedResult to match</param>
+        /// <returns>True if *any* 'What' property matches, False otherwise</returns>
+        public bool MatchesAnyWhat(ExtractedResult other)
+        {
+            return WhatComparer.MatchesWhat(this, other);
+        }
+
+        /// <summary>
+        ///  Match all of the 'Where' properties of two ExtractedResults (FileUri, StartLine/Column, EndLine/Column)
```

Comment:
>Column [](start = 109, length = 6)

Region matching is complicated. In addition to being specified by line/column properties, a region in a text file can also be specified by `CharOffset`+`CharLength`, and a region in a binary file is specified by `ByteOffset` +`ByteLength`. I will check the `WhereComparer` to make sure it does the right comparison.

Reply:
Verified that the `WhereComparer` compares all appropriate `Region` properties. Will clarify this comment.

---
In reply to: [296319818](https://github.com/microsoft/sarif-sdk/pull/1539#discussion_r296319818) [](ancestors = 296319818)